samples/boards/stm32/i2c_config/README.rst

```diff
@@ -0,0 +1,78 @@
+.. _i2c_config:
+
+STM32 i2c_config
+################
+
+Overview
+********
+This sample simply demonstrate the **get_config** API of the stm32 I2C driver.
+The I2C peripheral configuration is checked regarding the I2C bitrate which can be:
+
+	- <I2C_BITRATE_FAST_STANDARD>
+	- <I2C_BITRATE_FAST>
+	- <I2C_BITRATE_FAST_PLUS>
+
+
+In case of the I2C V2, the I2C peripheral of the STM32 microcontrollers have
+a TIMING register to write in order to generate the correct I2C clock signal.
+This value depends on the peripheral clock input and the I2C speed.
+The calculation of that TIMING value can be given by the `STM32CubeMX`_ application
+**or** by the present sample.
+
+Because the code sequence to calculate the I2C V2 TIMING value is heavy,
+the i2c_ll_stm32_v2.c driver should not embed it.
+That piece of code is activated by the CONFIG_I2C_STM32_V2_TIMING flag
+which must **not be set** except by this sample.
```

Comment:
```suggestion
Because the code sequence to calculate the I2C V2 TIMING value is heavy,
it is not advised to enable it in production binary.
By enabling CONFIG_I2C_STM32_V2_TIMING flag, this sample allows to 
retrieve timing register configuration for the defined ``clock-frequency``.
User can then configure timing in his application by adapting the following dts
snippet with the output of this sample:
```

Reply:
done